spd/eval.py

```diff
@@ -755,6 +756,165 @@ def compute(self) -> Mapping[str, float]:
         return {"loss/faithfulness": loss.item()}
 
 
+class GeometricSimilarityComparison(StreamingEval):
+    SLOW = True
+
+    def __init__(self, model: ComponentModel, config: Config, **kwargs: Any):
+        self.model = model
+        self.config = config
+        self.reference_run_path = kwargs.get("reference_run_path")
+        if self.reference_run_path is None:
+            raise ValueError("reference_run_path is required for GeometricSimilarityComparison")
+        self.kwargs = kwargs
+        self.reference_model: ComponentModel | None = None
+        self._computed_this_eval = False
+        self.device = next(iter(model.parameters())).device
+        self.n_tokens = 0
+        self.component_activation_counts: dict[str, Float[Tensor, " C"]] = {
+            module_name: torch.zeros(model.C, device=self.device)
+            for module_name in model.components
+        }
+
+    def _load_reference_model(self) -> ComponentModel:
```

Comment:
Another reason that I don't like doing this comparison during a run in the same process is because loading a reference model could be memory intensive and might make memory management during training more spiky (specifically, it may increase the amount of total gpu memory required for the run just to compute this eval).